storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransaction.java

```diff
@@ -160,6 +163,18 @@ public void removeProposerBoostRoot() {
     proposerBoostRootSet = true;
   }
 
+  @Override
+  public void addEquivocatingIndex(UInt64 equivocatingIndex) {
+    if (!store.equivocatingIndices.contains(equivocatingIndex)) {
+      equivocatingIndices.add(equivocatingIndex);
+    }
+  }
```

Comment:
How do we prune the set of equivocating indices? We only need to hold them temporarily because once the validator is marked as slashed new attestations are ignored - we're just trying to remove the weighting from previous attestations from fork choice. Since we can't process attestations more than 2 epochs after the slot they are for we only need to remember these indices for at most 2 epochs.

It may be better to not track this in `Store` but just in something like `ForkChoiceStrategy`.

Reply:
We also aren't storing these indices to disk so if we restart we'll forget them all.  That's probably not the end of the world but may be something we need to ask some questions about to properly understand the implications.

Reply:
The easiest approach may be to do this a completely different way:

* Remove the equivocating indices set entirely.
* Have two new fields in `VoteTracker` - `currentEquivocating` and `nextEquivocating`
* When a slashing is received, immediately update the `VoteTracker` for any slashed validators to mark it's "nextEquivocating" as true
* In `computeDeltas`
    * if `currentEquivocating` is true, we can skip processing deltas for this validator as we're already ignoring their vote
    * otherwise if `nextEquivocating` is true we treat the new balance as zero which will then calculate a delta that removes the weighting for this validator.  The `newVote` then has `currentEquivocating` set to true.

This avoids all the issues with managing the equivocating indices set and ensures equivocating validators are recorded on disk since changes to `VoteTracker` are always stored.